Il2CppInspector.Common/Reflection/TypeInfo.cs

```diff
@@ -94,13 +94,14 @@ private class TypeArgumentsComparer : EqualityComparer<TypeInfo[]>
 
         public List<MemberInfo> DeclaredMembers => new IEnumerable<MemberInfo>[] {
             DeclaredConstructors, DeclaredEvents, DeclaredFields, DeclaredMethods,
-            DeclaredNestedTypes?.ToList() ?? new List<TypeInfo>(), DeclaredProperties
+            DeclaredNestedTypes, DeclaredProperties
         }.SelectMany(m => m).ToList();
 
         public List<MethodInfo> DeclaredMethods { get; } = new List<MethodInfo>();
 
-        private readonly int[] declaredNestedTypes;
-        public IEnumerable<TypeInfo> DeclaredNestedTypes => declaredNestedTypes.Select(x => Assembly.Model.TypesByDefinitionIndex[x]);
+        // Note: DeclaredNestedTypes is deliberately empty on generic type instances.
```

Comment:
What does this commit do? I see the nested type indices are replaced with TypeRefs and the empty list checks are moved around but the net behaviour is the same?

Does the .NET Framework documentation say that DeclaredNestedTypes is meant to be empty for concrete generic types, or did you test and find this, or is that just the way it shakes out with our current implementation? I'm not seeing anything here that changes the way DeclaredNestedTypes is populated?

Reply:
DeclaredNestedTypes on concrete generic types, in the C# API, returns exactly the same output as it does on the generic definition, i.e. it outputs generic definitions without any of the type parameters substituted. This is not a surprise - nested types aren't actually required to inherit parameters from their containing types (and you can construct such types through IL).

I had thought that the nested types were empty for concrete generics, but that's not true in a test I just did - so either my testing was wrong or I had some other reason for making them empty. At this point, since I don't recall, I'm happy to change this to match the behaviour of having `DeclaredNestedTypes` return the nested types from the generic type definition.

Reply:
Feel free to go ahead and do that if you have the opportunity!

Reply:
Fixed (via rewriting this commit).